skopt/learning/gpr.py

```diff
@@ -0,0 +1,41 @@
+from sklearn.gaussian_process import GaussianProcessRegressor as sk_GaussianProcessRegressor
+from sklearn.gaussian_process.kernels import WhiteKernel
+
+class GaussianProcessRegressor(sk_GaussianProcessRegressor):
+    """
+    GaussianProcessRegressor that allows noise tunability.
+    """
+
+    def fit(self, X, y):
+        """Fit Gaussian process regression model
+
+        Parameters
+        ----------
+        X : array-like, shape = (n_samples, n_features)
+            Training data
+
+        y : array-like, shape = (n_samples, [n_output_dims])
+            Target values
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        super(GaussianProcessRegressor, self).fit(X, y)
+        for param, value in self.kernel_.get_params().items():
+            if param.endswith('noise_level'):
+                self.noise_ = value
+                break
+
+        # The noise component of this kernel should be set to zero
+        # while estimating K(X, X_test) and K(X_test, X_test)
+        # Note that the term K(X, X) should include the noise but
+        # this (K(X, X))^-1y is precomputed as the attribute alpha.
```

Comment:
Sorry I meant this `alpha_` (https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/gaussian_process/gpr.py#L229)